### PR TITLE
Webform normalizer

### PR DESCRIPTION
**Pull request by PavelBulat**

### CODE REVIEW

File: modules/custom/falcon_webform/src/Normalizer/WebformNormalizer.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+   * {@inheritdoc}
+   */
+  public function normalize($webform, $format = NULL, array $context = []) {
+    // Add the current webform entity as a cacheable dependency to make Drupal
+    // flush the cache when the webform entity gets updated.
+    $this->addCacheableDependency($context, $webform);
+
+    /* @see: \Drupal\webform\WebformEntityAccessControlHandler::checkAccess() */
+    if (!$webform->access('submission_page') || !$webform->isOpen()) {
+      // Do we need logger here??
```
Comment (kalabro) [inline modules/custom/falcon_webform/src/Normalizer/WebformNormalizer.php]: Yes, let's add a log record.

File: modules/custom/falcon_webform/src/Normalizer/WebformNormalizer.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      return [];
+    }
+
+    $normalized = [
+      'entity_type' => [['value' => $webform->getEntityTypeId()]],
+      'entity_bundle' => [['value' => $webform->bundle()]],
+      'id' => $webform->getOriginalId(),
+      'title' => $webform->label(),
+      'description' => $webform->getDescription(),
```
Comment (kalabro) [inline modules/custom/falcon_webform/src/Normalizer/WebformNormalizer.php]: To remove $webform->getDescription()

File: modules/custom/falcon_webform/src/Normalizer/WebformNormalizer.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      'success_message' => $webform->getSetting('confirmation_message'),
+    ];
+
+    // Get elements for build form.
+    $fields = $webform->getElementsDecoded();
+
+    // Normalize elements.
+    $normalizedFields = [];
+    foreach ($fields as $key => $field) {
+      $normalizedFields[] = array_merge($this->getNormalizedField($field), ['field_id' => $key]);
```
Comment (kalabro) [inline modules/custom/falcon_webform/src/Normalizer/WebformNormalizer.php]: ['field_id' => $key] + $this->getNormalizedField($field)

File: modules/custom/falcon_webform/src/Normalizer/WebformNormalizer.php [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      // Remove # from key.
+      if ($key[0] === "#") {
+        $result[substr($key, 1)] = $value;
+      }
+      // Add child if key has not # (Fieldset case).
+      else {
+        if (is_array($value)) {
+          $value = $this->getNormalizedField($value);
+        }
+        $children[] = array_merge($value, ['field_id' => $key]);
```
Comment (kalabro) [inline modules/custom/falcon_webform/src/Normalizer/WebformNormalizer.php]: `+`

File: modules/features/falcon_feature_webform/config/install/field.field.paragraph.webform.field_webform.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    - field.storage.paragraph.field_webform
+    - paragraphs.paragraphs_type.webform
+  module:
+    - webform
+id: paragraph.webform.field_webform
+field_name: field_webform
+entity_type: paragraph
+bundle: webform
+label: Webform
+description: 'Create new webform or edit existing webform you can on <a target="_blank" href="/admin/structure/webform">Webforms</a> page.'
```
Comment (kalabro) [inline modules/features/falcon_feature_webform/config/install/field.field.paragraph.webform.field_webform.yml]: To create a new webform, go to <a target="_blank" href="/admin/structure/webform">Webforms</a> page.

**Review by kalabro — CHANGES_REQUESTED** [review]

File: modules/custom/falcon_webform/src/Normalizer/WebformNormalizer.php (hunk 2 above)
Comment (PavelBulat) [reply modules/custom/falcon_webform/src/Normalizer/WebformNormalizer.php]: Removed.

File: modules/custom/falcon_webform/src/Normalizer/WebformNormalizer.php (hunk 1 above)
Comment (PavelBulat) [reply modules/custom/falcon_webform/src/Normalizer/WebformNormalizer.php]: Added.

File: modules/custom/falcon_webform/src/Normalizer/WebformNormalizer.php (hunk 3 above)
Comment (PavelBulat) [reply modules/custom/falcon_webform/src/Normalizer/WebformNormalizer.php]: Changed.

File: modules/custom/falcon_webform/src/Normalizer/WebformNormalizer.php (hunk 4 above)
Comment (PavelBulat) [reply modules/custom/falcon_webform/src/Normalizer/WebformNormalizer.php]: Changed.

File: modules/features/falcon_feature_webform/config/install/field.field.paragraph.webform.field_webform.yml (hunk 5 above)
Comment (PavelBulat) [reply modules/features/falcon_feature_webform/config/install/field.field.paragraph.webform.field_webform.yml]: Changed.

**Review by kalabro — APPROVED** [review]

**Pull request merged**